bundles/org.openhab.binding.venstarthermostat/src/main/java/org/openhab/binding/venstarthermostat/internal/handler/VenstarThermostatHandler.java

```diff
@@ -334,7 +356,31 @@ private void updateThermostat(int heat, int cool, VenstarSystemMode mode) {
             if (res.isSuccess()) {
                 log.debug("Updated thermostat");
                 // update our local copy until the next refresh occurs
-                infoData = new VenstarInfoData(cool, heat, infoData.getState(), mode);
+                infoData = new VenstarInfoData(cool, heat, infoData.getState(), mode, infoData.getAway());
+            } else {
+                log.debug("Failed to update thermostat: {}", res.getReason());
+                goOffline(ThingStatusDetail.COMMUNICATION_ERROR, "Thermostat update failed: " + res.getReason());
+            }
+        } catch (VenstarCommunicationException | JsonSyntaxException e) {
+            log.debug("Unable to fetch info data", e);
+            String message = e.getMessage();
+            goOffline(ThingStatusDetail.COMMUNICATION_ERROR, message != null ? message : "");
+        } catch (VenstarAuthenticationException e) {
+            goOffline(ThingStatusDetail.CONFIGURATION_ERROR, "Authorization Failed");
+        }
+    }
+
+    private void updateThermostatAway(VenstarAwayMode away) {
```

Comment:
This seems to duplicate the updateThermostat method,  would be cleaner if we could break this out and have a "updateControls" and "updateSettings" method that then in turn calls "updateThermostat" passing int the path string (setting or control) and the parameter map.

Reply:
Fair comment. I was also thinking there must be a cleaner way to do it. I'll give it a try.

Reply:
For future reference - the thermostat probably needs "updateControls", "updateSettings", "updateInfo", and possibly other functions as well, to replicate the thermostat local API. I'm planning on adding more functionality later, so there will be a few things to do here. I'll work on these things after this first "easy" (not easy :-)) piece of code gets included.